generator/src/_Azure.Generator.csproj

```diff
@@ -0,0 +1,27 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```

Comment:
Due to [IsClientLibrary](https://github.com/Azure/azure-sdk-for-net/blob/ab120de65ff8cdcc6f21c729a46b4eab19d861cc/eng/Directory.Build.Common.props#L14), the project can't start with `Azure`, put `_` in front for now, we need a better name for this.

Reply:
I revert the name back and tried to add below in csproj
```
	<PropertyGroup>
	  <IsClientLibrary>false</IsClientLibrary>
	</PropertyGroup>
```
I am still getting this error
> Azure.Generator.csproj : error NU1202: Package Microsoft.Generator.CSharp.ClientModel 1
.0.0-alpha.20240820.5 is not compatible with netstandard2.0 (.NETStandard,Version=v2.0). Package Microsoft.Generator.CSharp.Cl
ientModel 1.0.0-alpha.20240820.5 supports: net8.0 (.NETCoreApp,Version=v8.0)

Seems like it doesn't overwrite the central `IsClientLibrary`.

Reply:
override happens after root config added to MSConfig, so override more flags in csproj to workaround.

Reply:
Can you move the override to a local props file that after setting everything includes the root props?  this might fix the ordering the way we want it.

Reply:
Tried to add a local `Directory.Build.props` under `./package/npm/http-client-csharp`, even an empty one and with all the overrides in csproj still, I will get below error:
```
error NU1103: Unable to find a stable package Microsoft.Generator.CSharp.ClientModel with version
```

Reply:
If we add a local `Directory.Build.props`, it will take over the control from `eng/Directory.Build.Common.props`, and everything in it will be lost, I think that's not what we want.
Instead, deal with the change within `eng/Directory.Build.Common.props` with a new flag `IsGeneratorLibrary`.

Reply:
added local `Directory.Build.props`